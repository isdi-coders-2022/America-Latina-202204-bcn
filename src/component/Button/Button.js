import styled from "styled-components";

const PaginationButton = styled.button`
  width: 100px;
  background-color: white;
  color: black;
  border-radius: 20px;
  border-style: none;
  font-weight: bold;
  font-size: 20px;
`;

const ModifyInfoButton = styled.button`
  height: 33px;
  width: 185px;
  background-color: #adb69f;
  color: black;
  border-radius: 5px;
  border-style: none;
  font-size: 20px;
`;

const Button = ({ text, action, type }) => {
  return (
    <>
      {type === "pagination" && (
        <PaginationButton onClick={action}>{text}</PaginationButton>
      )}
      {type === "modifyInfo" && (
        <ModifyInfoButton onClick={action}>{text}</ModifyInfoButton>
      )}
    </>
  );
};

export default Button;
