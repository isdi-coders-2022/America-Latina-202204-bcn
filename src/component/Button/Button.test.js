import { render, screen } from "@testing-library/react";
import reactTestRenderer from "react-test-renderer";
import Button from "./Button";

describe("Given a Button component", () => {
  describe("When it receives a type pagination text containing 'Next Page'", () => {
    test("Then it should render the text 'Next Page' in a button", () => {
      const inputText = "Next Page";

      const tree = reactTestRenderer
        .create(<Button type="pagination" text={inputText} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
  describe("When it receives a type modifyInfo text containing 'Add to collection", () => {
    test("Then it should render the text in a button", () => {
      const inputText = "Add to collection";

      render(<Button type="modifyInfo" text={inputText} />);
      const searchedResult = screen.getByText(inputText);

      expect(searchedResult).toBeInTheDocument();
    });
  });
});
