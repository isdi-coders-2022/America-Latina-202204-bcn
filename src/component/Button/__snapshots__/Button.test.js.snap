// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Button component When it receives a type pagination text containing 'Next Page' Then it should render the text in a button 1`] = `
<button
  className="sc-bczRLJ jgMWzy"
>
  Next Page
</button>
`;
